starport/pkg/cosmosgen/templates/vuex/store/index.ts.tpl

```diff
@@ -92,55 +103,70 @@ export default {
+					{{ end }}
+				{{ else }}
+				if (query) {
+					value = (await queryClient.{{ camelCase .FullName }}({{ range $i,$a :=(index .Rules 0).Params}} key.{{$a}}, {{end}} query)).data
+				}else{
+					value = (await queryClient.{{ camelCase .FullName }}({{ range $i,$a :=(index .Rules 0).Params}}{{ if (gt $i 0)}}, {{ end}} key.{{$a}} {{end}})).data
+				}
+				{{ end }}
 				{{ if (index .Rules 0).HasQuery}}
 				while (all && (<any> value).pagination && (<any> value).pagination.nextKey!=null) {
-					let next_values=(await (await initQueryClient(rootGetters)).{{ camelCase .FullName }}({{ range $i,$a :=(index .Rules 0).Params}} key.{{$a}}, {{end}}{...query, 'pagination.key':(<any> value).pagination.nextKey})).data
-					for (let prop of Object.keys(next_values)) {
-						if (Array.isArray(next_values[prop])) {
-							value[prop]=[...value[prop], ...next_values[prop]]
-						}else{
-							value[prop]=next_values[prop]
-						}
-					}
+					let next_values=(await queryClient.{{ camelCase .FullName }}({{ range $i,$a :=(index .Rules 0).Params}} key.{{$a}}, {{end}}{...query, 'pagination.key':(<any> value).pagination.nextKey})).data
+					value = mergeResults(value,next_values);
 				}
 				{{ end }}
 				commit('QUERY', { query: '{{ .Name }}', key: { params: {...key}, query}, value })
 				if (subscribe) commit('SUBSCRIBE', { action: '{{ .FullName }}', payload: { options: { all }, params: {...key},query }})
 				return getters['get{{.Name }}']( { params: {...key}, query}) ?? {}
 			} catch (e) {
-				console.error(new SpVuexError('QueryClient:{{ .FullName }}', 'API Node Unavailable. Could not perform query.'))
+				let err = new SpVuexError('QueryClient:{{ .FullName }}', 'API Node Unavailable. Could not perform query.')
+				err.original = e
+				console.error(err)
 				return {}
 			}
 		},
 		{{ end }}
-		{{ range .Module.Msgs }}async send{{ .Name }}({ rootGetters }, { value, fee, memo }) {
+		{{ range .Module.Msgs }}async send{{ .Name }}({ rootGetters }, { value, fee = [], memo = '' }) {
 			try {
-				const msg = await (await initTxClient(rootGetters)).{{ camelCase .Name }}(value)
-				const result = await (await initTxClient(rootGetters)).signAndBroadcast([msg], {fee: { amount: fee, 
+				const txClient=await initTxClient(rootGetters)
+				const msg = await txClient.{{ camelCase .Name }}(value)
+				const result = await txClient.signAndBroadcast([msg], {fee: { amount: fee, 
   gas: "200000" }, memo})
 				return result
 			} catch (e) {
-				if (e.toString()=='wallet is required') {
+				if (e == MissingWalletError ) {
 					throw new SpVuexError('TxClient:{{ .Name }}:Init', 'Could not initialize signing client. Wallet is required.')
 				}else{
-					throw new SpVuexError('TxClient:{{ .Name }}:Send', 'Could not broadcast Tx.')
+					let err = new SpVuexError('TxClient:{{ .Name }}:Send', 'Could not broadcast Tx.')
+					err.original = e
```

Comment:
Is `original` directly printed to console when this error thrown? Otherwise it might be better to print it so, the error details would be more obvious to the user.

Reply:
AFAIK it is, I'll double check in any case